tests-e2e/core/snapshotenvironmentbinding_test.go

```diff
@@ -414,13 +414,11 @@ var _ = Describe("SnapshotEnvironmentBinding Reconciler E2E tests", func() {
 
 			// Check no GitOpsDeployment CR found with default name (longer name).
 			gitOpsDeployment := buildGitOpsDeploymentObjectMeta(gitOpsDeploymentName, binding.Namespace)
-			err = k8s.Get(&gitOpsDeployment, k8sClient)
-			Expect(apierr.IsNotFound(err)).To(BeTrue())
+			Consistently(gitOpsDeployment, "30s", "1s").ShouldNot(k8s.ExistByName(k8sClient), "wait 30s for the object not to exist")
 
-			// Check GitOpsDeployment is created with short name).
+			// Check GitOpsDeployment is created with short name.
 			gitOpsDeployment.Name = binding.Name + "-" + binding.Spec.Components[0].Name
-			err = k8s.Get(&gitOpsDeployment, k8sClient)
-			Expect(err).To(Succeed())
+			Eventually(gitOpsDeployment, "2m", "1s").Should(k8s.ExistByName(k8sClient))
```

Comment:
`gitOpsDeployment` should be changed to `& gitOpsDeployment`. I ran the test with this change and the tests passed after prefixing the `&`. Same change required in `L417` as well.